app/javascript/packages/form-steps/form-steps.tsx

```diff
@@ -135,7 +136,12 @@ interface FormStepsProps {
   /**
    * Callback triggered on step change.
    */
-  onStepChange?: () => void;
+  onStepChange?: (step: FormStep) => void;
+
+  /**
+   * Callback triggered on step submit.
+   */
+  onStepSubmit?: (stepName: string) => void;
```

Comment:
Since these callbacks are pretty similar, should we pass the same type of argument, either the string name or the step object?